src/core/styles/components/avatar/_avatar-theme.scss

```diff
@@ -2,29 +2,45 @@ $component-name: 'igx-avatar';
 
 $components: register-component($components, #{$component-name});
 
-@function igx-avatar-theme($icon-background: null, $image-background: null, $icon-color: null) {
+@function igx-avatar-theme(
+    $icon-background: null,
+    $icon-color: null,
+    $initials-background: null,
+    $initials-color: null,
+    $image-background: null
+) {
     @if not($icon-color) and $icon-background {
         $icon-color: text-contrast($icon-background);
     }
 
+    @if not($initials-color) and $initials-background {
+        $initials-color: text-contrast($initials-background);
+    }
+
     @return (
         icon-background: $icon-background,
         icon-color: $icon-color,
+        initials-background: $initials-background,
+        initials-color: $initials-color,
         image-background: $image-background
     );
 }
+
 @mixin igx-avatar($theme, $palette: $default-palette) {
     $default-theme: igx-avatar-theme(
-        $icon-background: igx-color($default-palette, grays, 400),
-        $icon-color: igx-color($default-palette, grays, 800),
+        $icon-background: igx-color($palette, 'grays', 400),
+        $icon-color: igx-color($palette, 'grays', 800),
+        $initials-background: igx-color($palette, 'grays', 400),
+        $initials-color: igx-color($palette, 'grays', 800),
         $image-background: transparent
     );
+
     $this: extend($default-theme, $theme);
+    $small-size: 40px;
+    $medium-size: 64px;
+    $large-size: 88px;
 
     %igx-avatar-display {
-        display: flex;
-        justify-content: center;
-        align-items: center;
         position: relative;
         pointer-events: none;
```

Comment:
Please restore pointer-events as the user may need to implement some logic based on them. Use user-select set to none to disallow text/icon selection.